### PR TITLE
Ability to change auth type 

### PR DESCRIPTION
**Pull request by afarid**
- Ability to change auth type as plain text is deprecated
- Enabling  service by default 
- Add auth_user per database

### CODE REVIEW

File: pgbouncer/values.yaml [hunk 1]
```diff
@@ -62,6 +84,14 @@ statsPeriod: 60
 # add key: value map entries to customSettings to be added to pbgouncer.ini [pgbouncer] section
 customSettings: {}
 
+# Service configuration
+service:
+  enabled: true
+  type: ClusterIP
+  name: ""
+  port: 5432
```
Comment (mattmessinger) [inline pgbouncer/values.yaml:92]: as mentioned above, moving the `port` here will cause problems for people who already have their own values overrides in existing installations / scripts. (even though this change is better)

File: pgbouncer/templates/service.yaml [hunk 2]
```diff
@@ -2,23 +2,20 @@
...
       {{ $key }}: {{ $value | quote }}
     {{- end }}
 spec:
   type: {{ .Values.service.type }}
   selector:
-    {{ template "pgbouncer.appLabel" . }}
+    {{- include "pgbouncer.selectorLabels" . | nindent 6 }}
   ports:
-    - port: {{ .Values.servicePort }}
+    - port: {{ .Values.service.port }}
```
Comment (mattmessinger) [inline pgbouncer/templates/service.yaml:17]: see comment below about changing servicePort.  We need to keep this as-is for backward compatibility.

**Review by mattmessinger — CHANGES_REQUESTED** [review]
Thank you for taking the time to clean up several places in this chart.  I'm glad you made it more "helm 3 standard".  There are a couple of places where changes were made that will impact existing installs, such as `servicePort`, that need to be addressed before we can accept this PR.

I also would like @matth-boise to take a look at the changes.

In general, excellent work, and thank you!

File: pgbouncer/Chart.yaml [hunk 3]
```diff
@@ -1,3 +1,6 @@
+apiVersion: v1
+appVersion: "1.0"
 description: PgBouncer chart
+home: https://github.com/cradlepoint/kubernetes-helm-chart-pgbouncer
 name: pgbouncer
-version: 1.0.10
+version: 1.0.11
```
[the remark this answers is not in the document]
Comment (matth-boise) [reply pgbouncer/Chart.yaml:6]: fyi, see https://github.com/cradlepoint/kubernetes-helm-chart-pgbouncer/blob/master/.github/workflows/main.yml#L35

File: pgbouncer/values.yaml [hunk 4]
```diff
@@ -1,39 +1,58 @@
...
 
 budget:
-  minAvailable: 0 # PodDisruptionBudget spec.minAvailable
+  # PodDisruptionBudget spec.minAvailable
+  minAvailable: 0
 
 image:
   repository: cradlepoint/pgbouncer
-  tag: latest
+  tag: 1.0.1
```
[the remark this answers is not in the document]
Comment (afarid) [reply pgbouncer/values.yaml:18]: Thanks!

File: examples/helmfile.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  - name: bitnami
+    url: https://charts.bitnami.com/bitnami
+
+releases:
+  - name: pgbouncer
+    namespace: default
+    chart: ../pgbouncer
+    values:
+    - users:
+        pgbouncer: md5bc3eb42ff2e374313252d5f9b8a023fa
```
[the remark this answers is not in the document]
Comment (afarid) [reply examples/helmfile.yaml]: This is done, I changed it to set me.

File: pgbouncer/templates/service.yaml (hunk 2 above)
Comment (afarid) [reply pgbouncer/templates/service.yaml]: This is done, it is using servicePort now.

File: pgbouncer/values.yaml [hunk 6]
```diff
@@ -1,39 +1,58 @@
+# For all pgbouncer config, please check https://www.pgbouncer.org/config.html
 # Deployment values for pgbouncer
 replicaCount: 1
 internalPort: 5432
-servicePort: 5432
 antiAffinity: soft
-nodeAffinity: {} # optionally define nodeAffinity
-tolerations: [] # optionally define tolerations
-nodeSelector: {} # optionally define nodeSelector
+# optionally define nodeAffinity
```
[the remark this answers is not in the document]
Comment (afarid) [reply pgbouncer/values.yaml]: Done, They are in line now.

File: pgbouncer/Chart.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (afarid) [reply pgbouncer/Chart.yaml]: Done, I reverted this back to original value.

File: pgbouncer/values.yaml [hunk 7]
```diff
@@ -1,39 +1,58 @@
+# For all pgbouncer config, please check https://www.pgbouncer.org/config.html
```
[the remark this answers is not in the document]
Comment (afarid) [reply pgbouncer/values.yaml:1]: I reverted all changes that will make existing chart to break. 

I introduced only new variables to make it more flexible like auth_user and auth_type

by the way, the default auth type (plain)  in the helm chart is deprecated, please check the documentation here  https://www.pgbouncer.org/config.html

File: pgbouncer/values.yaml [hunk 8]
```diff
@@ -42,15 +61,18 @@ connectionLimits:
   reservePoolTimeout: 5
 
 spec:
-    labels: {} # Add custom pod labels here    
-    annotations: {} # Add custom pod annotations here
-      # Annotations are converted to a JSON string and run through
-      # the helm `tpl` function. Then converted back to a map and
-      # rendered as YAML. This allows usage of go template directives
-      # in your annotations.
+  # Add custom pod labels here
```
[the remark this answers is not in the document]
Comment (afarid) [reply pgbouncer/values.yaml]: This should be fixed.

File: pgbouncer/templates/deployment-pgbouncer.yaml [hunk 9]
```diff
@@ -52,17 +40,12 @@ spec:
...
-                    {{ template "pgbouncer.appLabel" . }}
-                    release: {{ .Release.Name }}
+                    {{- include "pgbouncer.selectorLabels" . | nindent 20 }}
         {{- end }}
         {{- if .Values.nodeAffinity }}
         nodeAffinity:
           {{- toYaml .Values.nodeAffinity | nindent 10 }}
         {{- end }}
-      {{- if .Values.nodeSelector }}
-      nodeSelector:
```
Comment (mattmessinger) [inline pgbouncer/templates/deployment-pgbouncer.yaml:63]: why did you remove nodeSelector?  this could break existing deployments

File: pgbouncer/values.yaml [hunk 10]
```diff
@@ -1,39 +1,57 @@
+# For all pgbouncer config, please check https://www.pgbouncer.org/config.html
 # Deployment values for pgbouncer
 replicaCount: 1
 internalPort: 5432
 servicePort: 5432
 antiAffinity: soft
 nodeAffinity: {} # optionally define nodeAffinity
 tolerations: [] # optionally define tolerations
-nodeSelector: {} # optionally define nodeSelector
```
Comment (mattmessinger) [inline pgbouncer/values.yaml:8]: why was this removed?

**Review by mattmessinger — CHANGES_REQUESTED** [review]

**Review by matth-boise — CHANGES_REQUESTED** [review]
in general, great changes -- thanks so much!  i did find an `nindent` that must be removed.

File: pgbouncer/templates/_helpers.tpl [hunk 11]
```diff
@@ -16,8 +16,29 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
 {{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "pgbouncer.labels" -}}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+helm.sh/chart: {{ include "pgbouncer.chart" . }}
+{{- if .Values.labels }}
+{{- toYaml .Values.labels | nindent 4}}
```
[the remark this answers is not in the document]
Comment (afarid) [reply pgbouncer/templates/_helpers.tpl]: Thanks for clarification, it should be fixed now.

File: pgbouncer/values.yaml (hunk 10 above)
Comment (afarid) [reply pgbouncer/values.yaml:8]: sorry for this, I removed it as nodeSelector will be deprecated, as node affinity can express everything that nodeSelector can express.

File: pgbouncer/templates/deployment-pgbouncer.yaml (hunk 9 above)
Comment (afarid) [reply pgbouncer/templates/deployment-pgbouncer.yaml:63]: Fixed

**Comment by mattmessinger** [comment]
looks great @afarid !  @matth-boise are you good with me merging this?

File: pgbouncer/templates/_helpers.tpl (hunk 11 above)
[the remark this answers is not in the document]
Comment (matth-boise) [reply pgbouncer/templates/_helpers.tpl]: thx!

**Comment by matth-boise** [comment]
@mattmessinger yes - everthing looks good!
